### PR TITLE
Annotation of access control metadata to two clusters

### PR DESCRIPTION
**Pull request by hubTab**
#### Problem
Started annotating the XML cluster files with Access Control XML elements.
Normalized Normalized fabric_idx usage across the two updated cluster xml files.
- https://github.com/project-chip/connectedhomeip/issues/11204 (Define ACL cluster XML)
- https://github.com/project-chip/connectedhomeip/issues/11286 (Type fabric_idx is not used uniformly)

Issue #11204 
Issue #11286 
Issue #10246 

#### Change overview
Annotated the operational-credentials-cluster.xml and onoff-cluster.xml based on the Matter Specs, section 11.18, and appclusters, respectively.

- Normalized the <access> elements. Note that the writeable <attribute> attribute indicates if a field is writeable, or not. The <access> elements qualifies provileges on a per operation basis:
     * updated all <struct> elements to include <item> fieldId;
     * updated NodeOperationalCertStatus to include missing item elements;
     * updated NOCStruct to include a missing item element;

- Normalized fabric_idx usage across the two updated cluster xml files.

#### Testing
How was this tested? (at least one bullet point required)
* Not applicable, as it is a draft yet.

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/onoff-cluster.xml [hunk 1]
```diff
@@ -48,39 +48,71 @@ limitations under the License.
...
+    </attribute>
+    <attribute side="server" code="0x4003" define="START_UP_ON_OFF" type="enum8" writable="true" optional="true">
+      <description>StartUpOnOff</description>
+      <access op="read" privilege="view"/>
+      <access op="write" privilege="operate"/>
+    </attribute>
 
     <command source="client" code="0x00" name="Off" optional="false">
       <description>On receipt of this command, a device SHALL enter its ‘Off’ state. This state is device dependent, but it is recommended that it is used for power off or similar functions. On receipt of the Off command, the OnTime attribute SHALL be set to 0.</description>
+      <access op="invoke" privilege="view"/>
```
Comment (tcarmelveilleux) [inline src/app/zap-templates/zcl/data-model/chip/onoff-cluster.xml:79]: Please remove all the `<access op="invoke" privilege="view"/>`. 

The `op="invoke"` can only have 1 associated privilege

File: src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml [hunk 2]
```diff
@@ -19,32 +19,35 @@ limitations under the License.
...
+    <item fieldId="10" name="LabelConflict" value="0x0a"/>
+    <item fieldId="11" name="InvalidFabricIndex" value="0x0b"/>
   </enum>
 
   <struct name="NOCStruct">
     <cluster code="0x003E"/>
-    <item name="FabricIndex" type="INT8U"/>
-    <item name="NOC" type="OCTET_STRING"/>
+    <item fieldId="0" name="FabricIndex" type="fabric_idx"/>
+    <item fieldId="1" name="NOC" type="OCTET_STRING" length="400" isFabricSensitive="true"/>
```
Comment (tcarmelveilleux) [inline src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml:49]: Is `isFabricSensitive` an alias for the equivalent of an `access` tag?

File: src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml [hunk 3]
```diff
@@ -54,44 +57,71 @@ limitations under the License.
...
+      <access op="read" privilege="view"/>
+    </attribute>
+    <attribute side="server" code="0x0003" define="COMMISSIONED_FABRICS" type="INT8U" writable="false" optional="false">
+      <description>CommissionedFabrics</description>
+      <access op="read" privilege="view"/>
+    </attribute>
     <!-- 400 = 400 bytes for root cert -->
-    <attribute side="server" code="0x0004" define="TRUSTED_ROOTS" type="ARRAY" entryType="OCTET_STRING" length="400" writable="false" optional="false">TrustedRootCertificates</attribute>
-    <attribute side="server" code="0x0005" define="CURRENT_FABRIC_INDEX" type="fabric_idx" writable="false" optional="false">CurrentFabricIndex</attribute>
+    <attribute side="server" code="0x0004" define="TRUSTED_ROOTS" type="ARRAY" entryType="OCTET_STRING" length="400" writable="false" optional="false">
```
Comment (tcarmelveilleux) [inline src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml:79]: The 400 length for an array here doesn't make sense

File: src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml [hunk 4]
```diff
@@ -54,44 +57,71 @@ limitations under the License.
...
       <description>Sender is requesting attestation information from the receiver.</description>
       <arg name="AttestationNonce" type="OCTET_STRING"/>
+      <access op="invoke" privilege="administer"/>
     </command>
 
     <command source="server" code="0x01" name="AttestationResponse" optional="false">
       <description>An attestation information confirmation from the server.</description>
       <arg name="AttestationElements" type="OCTET_STRING"/>
       <arg name="Signature" type="OCTET_STRING"/>
+      <access op="invoke" privilege="administer"/>
```
Comment (tcarmelveilleux) [inline src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml:98]: There are no privileges for response commands

File: src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml [hunk 5]
```diff
@@ -54,44 +57,71 @@ limitations under the License.
...
     </command>
 
     <command source="client" code="0x02" name="CertificateChainRequest" optional="false">
```
Comment (tcarmelveilleux) [inline src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml:101]: Add `response="CertificateChainResponse"`

File: src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml [hunk 6]
```diff
@@ -101,39 +131,46 @@ limitations under the License.
...
       <arg name="ICACValue" type="OCTET_STRING" optional="true"/>
+      <access op="invoke" privilege="administer"/>
     </command>
 
     <command source="server" code="0x08" name="NOCResponse" optional="false">
       <description>Response to AddNOC or UpdateNOC commands.</description>
       <arg name="StatusCode" type="INT8U"/>
-      <arg name="FabricIndex" type="INT8U"/>
+      <arg name="FabricIndex" type="fabric_idx"/>
       <arg name="DebugText" type="CHAR_STRING"/>
```
Comment (tcarmelveilleux) [inline src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml:148]: Need the field IDs for these args

File: src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml (hunk 4 above)
Comment (hubTab) [reply src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml:98]: Very good observation.

File: src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml (hunk 6 above)
Comment (hubTab) [reply src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml:148]: As none of the current cluster xml files include a fieldId element in the <arg/> attributes, just to make sure I understand the requirement,  would this format work?

```
    <command source="server" code="0x08" name="NOCResponse" optional="false">
      <description>Response to AddNOC or UpdateNOC commands.</description>
      <arg fieldId="0" name="StatusCode" type="INT8U"/>
      <arg fieldId="1" name="FabricIndex" type="fabric_idx"/>
      <arg fieldId="2" name="DebugText" type="CHAR_STRING"/>
    </command>
```

File: src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml (hunk 3 above)
Comment (hubTab) [reply src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml:79]: This is what needs to be addressed as a group - pending on a group resolution :)
https://csamembers.slack.com/archives/G014G30SVV0/p1636739901137800?thread_ts=1636393190.014500&cid=G014G30SVV0

File: src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml (hunk 2 above)
Comment (hubTab) [reply src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml:49]: It has been suggested to use on Nov 1st: https://csamembers.slack.com/archives/C02FTN42KEE/p1635797498010800?thread_ts=1635795804.009300&cid=C02FTN42KEE.

File: src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml (hunk 5 above)
Comment (hubTab) [reply src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml:101]: Did the comment refer to the already existing response?

```
    <command source="server" code="0x03" name="CertificateChainResponse" optional="false">
      <description>A device attestation certificate (DAC) or product attestation intermediate (PAI) certificate from the server.</description>
      <arg name="Certificate" type="OCTET_STRING"/>
    </command>
```

File: src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml (hunk 2 above)
Comment (hubTab) [reply src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml:49]: This was suggested on Nov 1st: https://csamembers.slack.com/archives/C02FTN42KEE/p1635797498010800?thread_ts=1635795804.009300&cid=C02FTN42KEE

File: src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml (hunk 3 above)
Comment (hubTab) [reply src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml:79]: This is the topic that needs group resolution :)

File: src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml (hunk 6 above)
Comment (hubTab) [reply src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml:145]: None of the cluster xml files include a fieldId in the <arg/> attributes. To make sure I understand the requirement, is this format correct?

```
    <command source="server" code="0x08" name="NOCResponse" optional="false">
      <description>Response to AddNOC or UpdateNOC commands.</description>
      <arg fieldId="0" name="StatusCode" type="INT8U"/>
      <arg fieldId="1" name="FabricIndex" type="fabric_idx"/>
      <arg fieldId="2" name="DebugText" type="CHAR_STRING"/>
    </command>
```

Comment (hubTab) [reply src/app/zap-templates/zcl/data-model/chip/operational-credentials-cluster.xml:145]: The last commit made the above assumption. If incorrect, we can rollback the changes and work on determining the right format.

**Pull request closed without merging**